airbyte-integrations/connectors/source-mysql/src/test-integration/java/io/airbyte/integrations/source/mysql/CdcBinlogsMySqlSourceDatatypeTest.java

```diff
@@ -0,0 +1,171 @@
+  protected Database setupDatabase() throws Exception {
+    container = new MySQLContainer<>("mysql:8.0");
+    container.start();
+
+    config = Jsons.jsonNode(ImmutableMap.builder()
+        .put(JdbcUtils.HOST_KEY, container.getHost())
+        .put(JdbcUtils.PORT_KEY, container.getFirstMappedPort())
+        .put(JdbcUtils.DATABASE_KEY, container.getDatabaseName())
+        .put(JdbcUtils.USERNAME_KEY, container.getUsername())
+        .put(JdbcUtils.PASSWORD_KEY, container.getPassword())
+        .put("replication_method", MySqlSource.ReplicationMethod.CDC)
+        .build());
+
+    dslContext = DSLContextFactory.create(
+        config.get(JdbcUtils.USERNAME_KEY).asText(),
+        config.get(JdbcUtils.PASSWORD_KEY).asText(),
+        DatabaseDriver.MYSQL.getDriverClassName(),
+        String.format(DatabaseDriver.MYSQL.getUrlFormatString(),
+            config.get(JdbcUtils.HOST_KEY).asText(),
+            config.get(JdbcUtils.PORT_KEY).asInt(),
+            config.get(JdbcUtils.DATABASE_KEY).asText()),
+        SQLDialect.MYSQL);
+    final Database database = new Database(dslContext);
+
+    // It disable strict mode in the DB and allows to insert specific values.
+    // For example, it's possible to insert date with zero values "2021-00-00"
+    database.query(ctx -> ctx.fetch("SET @@sql_mode=''"));
+
+    revokeAllPermissions();
+    grantCorrectPermissions();
+
+    return database;
+  }
+
+  private void revokeAllPermissions() {
+    executeQuery("REVOKE ALL PRIVILEGES, GRANT OPTION FROM " + container.getUsername() + "@'%';");
+  }
+
+  private void grantCorrectPermissions() {
+    executeQuery(
+        "GRANT SELECT, RELOAD, SHOW DATABASES, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO "
+            + container.getUsername() + "@'%';");
+  }
+
+  private void executeQuery(final String query) {
+    try (final DSLContext dslContext = DSLContextFactory.create(
+        "root",
+        "test",
+        DatabaseDriver.MYSQL.getDriverClassName(),
+        String.format(DatabaseDriver.MYSQL.getUrlFormatString(),
+            container.getHost(),
+            container.getFirstMappedPort(),
+            container.getDatabaseName()),
+        SQLDialect.MYSQL)) {
+      final Database database = new Database(dslContext);
+      database.query(
+          ctx -> ctx
+              .execute(query));
+    } catch (final Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public boolean testCatalog() {
+    return true;
+  }
+
+  @Override
+  protected void addTimestampDataTypeTest() {
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("timestamp")
+            .airbyteType(JsonSchemaType.STRING_TIMESTAMP_WITH_TIMEZONE)
+            .addInsertValues("null", "'2021-01-00'", "'2021-00-00'", "'0000-00-00'", "'2022-08-09T10:17:16.161342Z'")
+            .addExpectedValues(null, "1970-01-01T00:00:00.000000Z", "1970-01-01T00:00:00.000000Z", "1970-01-01T00:00:00.000000Z",
```

Comment:
is 1970-01-01 the equivalent of the `null` entries in the default test cases? (adding this as a comment would be helpful)